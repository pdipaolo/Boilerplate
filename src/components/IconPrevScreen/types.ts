type PropsTypes = {
  canGoBack?: boolean;
};

export { type PropsTypes };
