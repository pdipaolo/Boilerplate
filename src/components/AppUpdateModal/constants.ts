const APP_UPDATE_TYPE = {
  SOFT: 1 as 1,
  HARD: 2 as 2,
};

export { APP_UPDATE_TYPE };
