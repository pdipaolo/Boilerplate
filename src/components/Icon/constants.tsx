import React from 'react';

import { type SvgProps } from 'react-native-svg';

import Bonus from '../../assets/svg/bonus.svg';

const ICON = {
  // General
  BONUS: 'bonus' as 'bonus',
};

const ICON_COMPONENT = {
  // General
  BONUS: {
    ID: ICON.BONUS,
    COMPONENT: (props: SvgProps) => <Bonus {...props} />,
  },
};

function findIconById(
  id: (typeof ICON)[keyof typeof ICON] | undefined,
  props: SvgProps & {
    primaryColor: string;
    secondaryColor?: string;
    tertiaryColor?: string;
  },
): JSX.Element | undefined {
  const ICONS = Object.values(ICON_COMPONENT);
  const CURR_ICON = ICONS.find((obj) => obj.ID === id);
  return CURR_ICON?.COMPONENT(props);
}

export { ICON, findIconById };
