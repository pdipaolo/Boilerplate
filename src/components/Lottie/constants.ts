import LoaderLottie from '../../assets/animations/loader.json';

const ANIMATION = {
  LOADER: 'loader' as 'loader',
};

const ANIMATION_DATA = {
  [ANIMATION.LOADER]: LoaderLottie,
};

export { ANIMATION, ANIMATION_DATA };
