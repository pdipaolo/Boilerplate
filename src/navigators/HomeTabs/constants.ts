import tab1Route from '../../screens/Tab1/route';
import tab2Route from '../../screens/Tab2/route';
import tab3Route from '../../screens/Tab3/route';
import tab4Route from '../../screens/Tab4/route';
import tab5Route from '../../screens/Tab5/route';

const bottomRoutes = [tab1Route, tab2Route, tab3Route, tab4Route, tab5Route];

export { bottomRoutes};
