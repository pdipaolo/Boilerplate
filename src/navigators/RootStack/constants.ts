import createAccountRoute from '../../screens/CreateAccount/route';
import loginRoute from '../../screens/Login/route';

const routes = [loginRoute, createAccountRoute];

export { routes };
