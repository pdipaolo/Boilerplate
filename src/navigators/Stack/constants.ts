import { createStackNavigator } from '@react-navigation/stack';

import bottomTabsRoute from '../../screens/BottomTabs/route';
import introRoute from '../../screens/Intro/route';

import { type StackParamsList } from './types';

const Stack = createStackNavigator<StackParamsList>();
const StackNavigator = Stack.Navigator;
const StackScreen = Stack.Screen;

const routes = [
  introRoute,
  bottomTabsRoute,
];

export { StackNavigator, StackScreen, routes };
