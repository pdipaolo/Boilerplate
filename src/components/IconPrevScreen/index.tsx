import React from 'react';

import { Pressable } from 'react-native';

import { useNavigation } from '../../navigators/Stack/hooks';
import Icon from '../Icon';
import { ICON } from '../Icon/constants';

import styles from './styles';
import { type PropsTypes } from './types';

function IconPrevScreen(props: PropsTypes): JSX.Element {
  const { canGoBack } = props;

  const navigation = useNavigation();

  const handleOnPress = () => {
    navigation.goBack();
  };

  if (!canGoBack) {
    return <></>;
  }

  return (
    <Pressable style={styles.container} onPress={handleOnPress}>
      <Icon id={ICON.CHEVRON_LEFT} />
    </Pressable>
  );
}

export default IconPrevScreen;
