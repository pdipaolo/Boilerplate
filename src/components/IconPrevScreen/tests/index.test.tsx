import React from 'react';

import { render } from '@testing-library/react-native';

import IconPrevScreen from '../';
import { ICON } from '../../Icon/constants';

describe('IconPrevScreen', () => {
  it('should match snapshot', () => {
    const screen = render(<IconPrevScreen canGoBack />);

    expect(screen).toMatchSnapshot();
  });

  describe('pressable style', () => {
    it('should have padding: 10', () => {
      const { root } = render(<IconPrevScreen canGoBack />);

      const received = root.props.style.padding;
      const expected = 10;

      expect(received).toBe(expected);
    });
  });

  // TODO: add navigation test

  describe('icon', () => {
    it('should have icon', () => {
      const { root } = render(<IconPrevScreen canGoBack />);

      const icon = root.props.children;
      const received = icon;

      expect(received).toBeDefined();
    });

    describe('icon props', () => {
      it(`should have id: ${ICON.CHEVRON_LEFT}`, () => {
        const { root } = render(<IconPrevScreen canGoBack />);

        const icon = root.props.children[0];

        const received = icon.props.id;
        const expected = ICON.CHEVRON_LEFT;

        expect(received).toBe(expected);
      });
    });
  });
});
