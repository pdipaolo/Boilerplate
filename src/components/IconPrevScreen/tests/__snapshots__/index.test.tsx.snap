// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconPrevScreen should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "hotpink",
      "padding": 10,
    }
  }
>
  <SvgMock
    primaryColor="#212121"
  />
</View>
`;
